File: solutions/1379/c++/solution.cpp
// 1379. Find a Corresponding Node of a Binary Tree in a Clone of That Tree
class Solution
{
	public:
	TreeNode* getTargetCopy(TreeNode* original, TreeNode* cloned, TreeNode* target)
	{
		if (original == nullptr)
		{
			return nullptr;
		}

		if (original == target)
		{
			return cloned;
		}

		TreeNode* result = getTargetCopy(original->left, cloned->left, target);

		if (result != nullptr)
		{
			return result;
		}

		return getTargetCopy(original->right, cloned->right, target);
	}
};

File: solutions/1379/js/solution.js
// 1379. Find a Corresponding Node of a Binary Tree in a Clone of That Tree
var getTargetCopy = function(original, cloned, target)
{
	let originalStack = new Array();
	let clonedStack = new Array();

	originalStack.push(original);
	clonedStack.push(cloned);

	while (originalStack.length > 0)
	{
		let originalNode = originalStack.pop();
		let clonedNode = clonedStack.pop();

		if (originalNode == null)
		{
			continue;
		}

		if (originalNode === target)
		{
			return clonedNode;
		}

		originalStack.push(originalNode.left);
		originalStack.push(originalNode.right);

		clonedStack.push(clonedNode.left);
		clonedStack.push(clonedNode.right);
	}

	return null;
};